apps/src/templates/Dialog.jsx

```diff
@@ -82,23 +85,34 @@ Cancel.propTypes = {
   children: React.PropTypes.node,
 };
 
-export function Buttons({children}) {
-  children = React.Children.toArray(children);
-  var leftChildren = children.slice(0, children.length-1);
-  var rightChild = children[children.length - 1];
-  return (
-    <div style={styles.buttons}>
-      <div style={styles.rightButton}>
-        {rightChild}
+export const Buttons = React.createClass({
+  getConfirmButton() {
+    return this.refs.buttons.getElementsByClassName('confirmButton')[0];
+  },
+
+  getCancelButton() {
+    return this.refs.buttons.getElementsByClassName('cancelButton')[0];
+  },
```

Comment:
I'm not worried about accidentally selecting the wrong elements, more that it opens the door to future abuse, like someone adding global css rules for `.confirmButton` and `.cancelButton`, thus creating an unintentional coupling between styling (whatever the css rules say) and semantic behavior (hitting tab toggles focus between the buttons).

To "protect" against this, you could make the class name something like "_cancelButtonDoNotUseForStylingOrYouWillBeFired" but that's kind of ugly 😄